src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -1269,23 +1269,19 @@ internal static string ToStringParser(ExecutionContext context, object obj, IFor
         /// </exception>
         internal static string ToString(ExecutionContext context, object obj, string separator, string format, IFormatProvider formatProvider, bool recurse, bool unravelEnumeratorOnRecurse)
         {
-            PSObject mshObj = obj as PSObject;
-
-            #region plain object
-            if (mshObj == null)
+            bool TryFastTrackPrimitiveTypes(object obj, out string str)
```

Comment:
Is this a new nested method syntax? I'm in two minds about it; other than making the body of this method noisier, I'm not sure what it offers beyond a normal `private` method. For example, I don't see why the JIT would be less able to inline the latter.

I can also imagine this method having broader usage in `LanguagePrimitives`.

Reply:
It's a local function.

> I'm not sure what it offers beyond a normal private method.

C# make it more efficient to access the variables from the enclosing method via `ref` reference (vs. parameter argument passed to a private method). Here, I do access the `formatProvider` variable directly within the local function, but there is no much difference from passing `formatProvider` to a private method. This logic was scoped only to this method, so I tend to keep it that way until it actually gets used elsewhere.

Reply:
Renamed the `obj` parameter to `value`.